lib/cstruct.ml

```diff
@@ -88,10 +88,15 @@ let of_bigarray ?(off=0) ?len buffer =
 let to_bigarray buffer =
   Bigarray.Array1.sub buffer.buffer buffer.off buffer.len
 
-let create len =
+let create_unsafe len =
   let buffer = Bigarray.(Array1.create char c_layout len) in
   { buffer ; len ; off = 0 }
 
+let create len =
+  let t = create_unsafe len in
+  Bigarray.Array1.fill t.buffer (Char.chr 0);
```

Comment:
consistency is my argument (I've to keep less symbols in my head when reading the code)